pool.go

```diff
@@ -186,8 +186,14 @@ func (p *Pool) recreateConnection(closedConn *Connection) {
 		}
 
 		p.handleError(fmt.Errorf("failed to reconnect to %s: %w", conn.addr, err))
+
 		select {
-		case <-time.After(p.Opts.ReconnectWait):
+		case <-time.After(reconnectTime):
+			reconnectTime *= 2
+			if reconnectTime > p.Opts.MaxReconnectWait {
+				// stop connection retry after MaxReconnectWait
```

Comment:
we can continue to set `reconnectTime` to `MaxReconnectTime` without returning from the re-try loop.

Reply:
As I understand, we don't want to stop connection retry ever, once we reach  MaxReconnectTime, we can stop increasing reconnectTime, and keep retrying every MaxReconnectTime duration.

The changes as per this are done.